File: src/ch3/bmi_struct.py
# 키와 몸무게 입력
height_cm = float(input('키 (cm) : '))
weight = float(input('몸무게(kg) : '))

# BMI 계산
height = height_cm / 100.0
bmi = weight / (height ** 2)

# 비만도 판정표 - 판전용 사전 타입 리스트
bmi_list = [
    {"min": 0, "max": 18.5, "label": "저체중"},
    {"min": 18.5, "max": 25, "label": "정상"},
    {"min": 25, "max": 30, "label": "비만전단계"},
    {"min": 30, "max": 35, "label": "1단계 비만"},
    {"min": 35, "max": 40, "label": "2단계 비만"},
    {"min": 40, "max": 99, "label": "3단계 비만"}]

# 판정
result = "계산 불가"
for range in bmi_list:
    if range["min"] <= bmi < range["max"]:
        result = range["label"]

# 결과 표시
print("BMI = {:.1f}, 비만도 = {}".format(bmi, result))

File: src/ch3/bmi_struct.rs
// BMI 판정용 구조체 --- (*1)
struct BmiRange {
    min: f64,            // min이상
    max: f64,            // max미만
    label: &'static str, // 판정
}
fn main() {
    // 키와 몸무게 입력\ --- (*2)
    let height_cm = input("키(cm) : ");
    let weight = input("몸무게(kg) : ");
    // BMI 계산
    let height = height_cm / 100.0;
    let bmi = weight / height.powf(2.0);
    // 비만도 판정표를 벡터 타입으로 생성 --- (*3)
    let bmi_list = vec![
        BmiRange {
            min: 0.0,
            max: 18.5,
            label: "저체중",
        },
        BmiRange {
            min: 18.5,
            max: 23.0,
            label: "정상",
        },
        BmiRange {
            min: 23.0,
            max: 25.0,
            label: "비만전단계",
        },
        BmiRange {
            min: 25.0,
            max: 30.0,
            label: "1단계 비만",
        },
        BmiRange {
            min: 30.0,
            max: 35.0,
            label: "2단계 비만",
        },
        BmiRange {
            min: 35.0,
            max: 99.0,
            label: "3단계 비만",
        },
    ];
    // 비만도 판단 --- (*4)
    let mut result = "계산 불가";
    for range in bmi_list {
        if range.min <= bmi && bmi < range.max {
            result = range.label;
            break;
        }
    }
    // 결과 표시
    println!("BMI = {:.1}, 비만도 = {}", bmi, result);
}
// 한 줄씩 읽어 f64 타입으로 반환 --- (*5)
fn input(prompt: &str) -> f64 {
    println!("{}", prompt);
    let mut s = String::new();
    std::io::stdin().read_line(&mut s).expect("입력 에러");
    s.trim().parse().expect("숫자 변환 에러")
}
